waspc/waspls/test/Wasp/LSP/CompletionTest.hs

```diff
@@ -0,0 +1,145 @@
+module Wasp.LSP.CompletionTest where
+
+import Control.Lens ((^.))
+import Control.Monad.Log.Pure (runLog)
+import Control.Monad.State.Strict (evalStateT)
+import qualified Data.ByteString.Lazy as BS
+import qualified Data.ByteString.Lazy.Char8 as BSC
+import Data.Foldable (find)
+import Data.List (isPrefixOf)
+import Data.Maybe (mapMaybe)
+import qualified Language.LSP.Types as LSP
+import qualified Language.LSP.Types.Lens as LSP
+import System.FilePath (replaceExtension, takeBaseName)
+import Test.Tasty (TestTree, testGroup)
+import Test.Tasty.Golden (findByExtension, goldenVsStringDiff)
+import Wasp.Analyzer.Parser.ConcreteParser (parseCST)
+import qualified Wasp.Analyzer.Parser.Lexer as Lexer
+import Wasp.LSP.Completion (getCompletionsAtPosition)
+import Wasp.LSP.ServerState (ServerState (ServerState, _cst, _currentWaspSource, _latestDiagnostics))
+
+-- | Run test cases in ./completionTests directory
+--
+-- See 'readCompletionTest' for the format of a test case.
+test_CompletionLists :: IO TestTree
+test_CompletionLists = do
+  inputFiles <- findByExtension [".wasp"] "./waspls/test/Wasp/LSP/completionTests"
+  return $
+    testGroup "Wasp.LSP.Completion.getCompletionsAtPosition" $
+      makeCompletionTest <$> inputFiles
+
+-- | Takes a completion test case and produces the list of completion items
```

Comment:
When somebody says "test case", I imagine both input of test and expected result. So I wouldn't use "test case" here, but instead "completion test input". Or "wasp source with cursor". I suggested below to introduce either `type` or `newtype` for it so that should help.

Reply:
What is also a bit confusing here is that you say it produces a list of completion items, but I see it produces a `String`. Ok, I am guessing that list of completion items is encoded as a `String` then, but still, I wonder a bit if this is a mistake in documentation. So maybe we can explain why is it just a String.